src/CatSearch.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from "react"
+import { useState, useEffect, useCallback } from "react"
+import { fetchSearchResultsFromAPI } from "./searchApiClient"
+
+export function CatSearch() {
+  const [valid, setValid] = useState(false)
+  const [hits, setHits] = useState<any[]>([])
+  const [searchTerm, setSearchTerm] = useState("")
+
+  const searchTermOnChange = useCallback((event) => {
+    const { value } = event.target
+    setSearchTerm(value)
+  }, [])
+
+  useEffect(() => {
+    setValid(true)
+    if (searchTerm.length < 2) {
+      setValid(false)
+      return
+    }
+
+    fetchSearchResultsFromAPI(searchTerm).then((hits) => {
+      setHits(hits)
+    })
+  })
+
+  return (
+    <div>
+      <h2>Search for cat breed</h2>
+      <input type="text" onChange={searchTermOnChange} />
+      <div>
+        {valid === false
+          ? "Type at least two characters to start searching"
+          : `You searched for ${searchTerm}`}
+      </div>
+      <div>
+        {(hits && hits.length) > 0
```

Comment:
There is actually a potential bug here. Since the `> 0` is outside the parentheses, if `hits` is ever falsey, our assertion will try to see if a falsey value like null or undefined will be greater than 0.

So, we can move that comparison inside the parenthesis so that we're only ever comparing `hits.length > 0`.

But, we can actually just remove the `> 0` and get the same result. Since 0 is a falsey value, if the length of the array is 0 `hits && hits.length` will return false automagically for us.